tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/IntegrationMapper.cs

```diff
@@ -12,889 +13,945 @@
                     }
-                }
-                else if (parameterProxyType == null)
-                {
-                    WriteLoadArgument(ilWriter, i, mustLoadInstance);
+
+                    callGenericTypes.Add(sourceParameterTypeElementType);
                 }
                 else
                 {
-                    ThrowHelper.ThrowInvalidCastException($"DuckType constraints is not supported on ByRef parameters. The target parameter {targetParameterType} can't be assigned from {sourceParameterType}");
+                    ThrowHelper.ThrowInvalidCastException($"DuckType constraints cannot be used in ByRef arguments. ({targetParameterTypeConstraint})");
                 }
             }
-
-            // Call method
-            onMethodBeginMethodInfo = onMethodBeginMethodInfo.MakeGenericMethod(callGenericTypes.ToArray());
-            ilWriter.EmitCall(OpCodes.Call, onMethodBeginMethodInfo, null);
-            ilWriter.Emit(OpCodes.Ret);
-
-            Log.Debug("Created BeginMethod Dynamic Method for '{IntegrationType}' integration. [Target={TargetType}]", integrationType.FullName, targetType.FullName);
-            return callMethod;
-        }
-
-        internal static DynamicMethod CreateSlowBeginMethodDelegate(Type integrationType, Type targetType)
-        {
-            /*
-             * OnMethodBegin signatures with 1 or more parameters with 1 or more generics:
-             *      - CallTargetState OnMethodBegin<TTarget>(TTarget instance);
-             *      - CallTargetState OnMethodBegin<TTarget, TArg1>(TTarget instance, TArg1 arg1);
-             *      - CallTargetState OnMethodBegin<TTarget, TArg1, TArg2>(TTarget instance, TArg1 arg1, TArg2);
-             *      - CallTargetState OnMethodBegin<TTarget, TArg1, TArg2, ...>(TTarget instance, TArg1 arg1, TArg2, ...);
-             *      - CallTargetState OnMethodBegin<TTarget>();
-             *      - CallTargetState OnMethodBegin<TTarget, TArg1>(TArg1 arg1);
-             *      - CallTargetState OnMethodBegin<TTarget, TArg1, TArg2>(TArg1 arg1, TArg2);
-             *      - CallTargetState OnMethodBegin<TTarget, TArg1, TArg2, ...>(TArg1 arg1, TArg2, ...);
-             *
-             */
-
-            Log.Debug("Creating SlowBeginMethod Dynamic Method for '{IntegrationType}' integration. [Target={TargetType}]", integrationType.FullName, targetType.FullName);
-            MethodInfo onMethodBeginMethodInfo = integrationType.GetMethod(BeginMethodName, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static);
-            if (onMethodBeginMethodInfo is null)
+            else
             {
-                Log.Debug("'{BeginMethodName}' method was not found in integration type: '{IntegrationType}'.", BeginMethodName, integrationType.FullName);
-                return null;
-            }
+                var srcParameterType = sourceParameterType.IsByRef ? sourceParameterType.GetElementType()! : sourceParameterType;
```

Comment:
You assumed `sourceParameterType.GetElementType()` could return `null` in the previous block, but not here? 🤔

Reply:
because `GetElementType()` doesn't return null if `IsByRef` is true, that's the way we get the real type without the ByRef stuff (ref, out, in)